File: app/src/main/java/com/teamunemployment/lolanalytics/FrontPage/BaseActivityPresenter.java
package com.teamunemployment.lolanalytics.FrontPage;

import android.util.Log;

import com.aurelhubert.ahbottomnavigation.AHBottomNavigation;
import com.aurelhubert.ahbottomnavigation.AHBottomNavigationItem;
import com.ncapdevi.fragnav.FragNavController;
import com.teamunemployment.lolanalytics.R;

import javax.inject.Inject;

/**
 * @author Josiah Kendall
 */

public class BaseActivityPresenter implements BaseActivityContract.Presenter {

    private BaseActivityView view;
    private static final int TOP = 0;
    private static final int JUNGLE = 1;
    private static final int MID = 2;
    private static final int MARKSMAN = 3;
    private static final int SUPPORT = 4;

    private BaseActivityPersistenceInteractor baseActivityPersistenceInteractor;

    private AHBottomNavigation bottomBar;

    @Inject
    public BaseActivityPresenter(BaseActivityPersistenceInteractor baseActivityPersistenceInteractor) {
        this.baseActivityPersistenceInteractor = baseActivityPersistenceInteractor;
    }

    public void setView(BaseActivityView view) {
        this.view = view;
    }

    @Override
    public void start() {

    }

    @Override
    public void handleError(Throwable e) {
        Log.e("BaseActivityPresenter", "Error : " + e.getMessage());
    }

    @Override
    public void restart() {

    }

    @Override
    public void resume() {

    }

    @Override
    public void pause() {

    }

    @Override
    public void stop() {

    }

    @Override
    public void destroy() {

    }


    @Override
    public void handleTabPress(int tab) {

        long summonerId = -1; // TODO
        switch (tab) {
            case TOP:
                baseActivityPersistenceInteractor.fetchWinRateForRole(-1,"TOP",this);
                view.setRoleName("Top");
                view.setTabIcon(R.drawable.top);
                view.setCorrectTabFragment(FragNavController.TAB1);
                break;
            case JUNGLE:
                baseActivityPersistenceInteractor.fetchWinRateForRole(-1,"JUNGLE",this);
                view.setRoleName("Jungle");
                view.setTabIcon(R.drawable.jg);
                view.setCorrectTabFragment(FragNavController.TAB2);
                break;
            case MID:
                baseActivityPersistenceInteractor.fetchWinRateForRole(-1,"MID",this);
                view.setRoleName("Mid");
                view.setTabIcon(R.drawable.mid);
                view.setCorrectTabFragment(FragNavController.TAB3);
                break;
            case MARKSMAN:
                baseActivityPersistenceInteractor.fetchWinRateForRole(-1,"MARKSMAN",this);
                view.setRoleName("Marksman");
                view.setTabIcon(R.drawable.bot);
                view.setCorrectTabFragment(FragNavController.TAB4);
                break;
            case SUPPORT:
                baseActivityPersistenceInteractor.fetchWinRateForRole(-1,"PLAYMAKER",this);
                view.setRoleName("Support");
                view.setTabIcon(R.drawable.sup);
                view.setCorrectTabFragment(FragNavController.TAB5);
                break;
        }
    }

    @Override
    public void onWinRateLoaded(double winRate) {
        view.setWinRate(winRate);
    }

    @Override
    public void setUpTabFragments() {
        // Do fragment setup

    }

    @Override
    public void setUpBottomBar(AHBottomNavigation bottomBar) {
        this.bottomBar = bottomBar;


        // Bottom bar items.
        AHBottomNavigationItem topItem = new AHBottomNavigationItem("Top", R.drawable.top);
        AHBottomNavigationItem jgItem = new AHBottomNavigationItem("Jungle", R.drawable.jg);
        AHBottomNavigationItem midItem = new AHBottomNavigationItem("Mid", R.drawable.mid);
        AHBottomNavigationItem adcItem = new AHBottomNavigationItem("Bottom", R.drawable.bot);
        AHBottomNavigationItem supItem = new AHBottomNavigationItem("Support", R.drawable.sup);

        // Add items to bottom bar.
        bottomBar.addItem(topItem);
        bottomBar.addItem(jgItem);
        bottomBar.addItem(midItem);
        bottomBar.addItem(adcItem);
        bottomBar.addItem(supItem);


        // Click listener for the bottom bar buttons.
        bottomBar.setOnTabSelectedListener(new AHBottomNavigation.OnTabSelectedListener() {

            @Override
            public boolean onTabSelected(int position, boolean wasSelected) {
                handleTabPress(position);
                return true;
            }
        });
    }

}

File: app/src/main/java/com/teamunemployment/lolanalytics/front_page/BaseActivityPresenter.kt
package com.teamunemployment.lolanalytics.front_page

import android.util.Log

import com.aurelhubert.ahbottomnavigation.AHBottomNavigation
import com.aurelhubert.ahbottomnavigation.AHBottomNavigationItem
import com.ncapdevi.fragnav.FragNavController
import com.teamunemployment.lolanalytics.R
import com.teamunemployment.lolanalytics.Utils.Role


/**
 * @author Josiah Kendall
 */

class BaseActivityPresenter
constructor(private val baseActivityPersistenceInteractor: BaseActivityPersistenceInteractor) : BaseActivityContract.Presenter {

    private lateinit var view : BaseActivityView

    private var bottomBar: AHBottomNavigation? = null

    fun setView(view: BaseActivityView) {
        this.view = view
    }

    override fun start() {

    }

    override fun showMessage(message: String) {
        view.showMessage(message)
    }

    override fun handleError(e: Throwable) {
        Log.e("BaseActivityPresenter", "Error : " + e.message)
    }

    override fun restart() {

    }

    override fun resume() {

    }

    override fun pause() {

    }

    override fun stop() {

    }

    override fun destroy() {

    }


    override fun handleTabPress(tab: Int) {

        val summonerId : String = "-1" // TODO
        when (tab) {
            TOP -> {
                baseActivityPersistenceInteractor.fetchWinRateForRole("-1"/*todo*/, "TOP", "oce", this)
                view!!.setCorrectTabFragment(FragNavController.TAB1)
            }
            JUNGLE -> {
                baseActivityPersistenceInteractor.fetchWinRateForRole("-1", "JUNGLE", "oce", this) // todo
                view!!.setCorrectTabFragment(FragNavController.TAB2)
            }
            MID -> {
                baseActivityPersistenceInteractor.fetchWinRateForRole("-1", "MID", "oce", this)
                view!!.setCorrectTabFragment(FragNavController.TAB3)
            }
            MARKSMAN -> {
                baseActivityPersistenceInteractor.fetchWinRateForRole("-1", "MARKSMAN", "oce", this)
                view!!.setCorrectTabFragment(FragNavController.TAB4)
            }
            SUPPORT -> {
                baseActivityPersistenceInteractor.fetchWinRateForRole("-1", "PLAYMAKER", "oce", this)
                view!!.setCorrectTabFragment(FragNavController.TAB5)
            }
        }
    }

    override fun onWinRateLoaded(winRate: Double) {

    }

    override fun setUpTabFragments() {
        // Do fragment setup

    }

    override fun setUpBottomBar(bottomBar: AHBottomNavigation) {
        this.bottomBar = bottomBar


        // Bottom bar items.
        val topItem = AHBottomNavigationItem("Top", R.drawable.top)
        val jgItem = AHBottomNavigationItem("Jungle", R.drawable.jg)
        val midItem = AHBottomNavigationItem("Mid", R.drawable.mid)
        val adcItem = AHBottomNavigationItem("Bottom", R.drawable.bot)
        val supItem = AHBottomNavigationItem("Support", R.drawable.sup)

        // Add items to bottom bar.
        bottomBar.addItem(topItem)
        bottomBar.addItem(jgItem)
        bottomBar.addItem(midItem)
        bottomBar.addItem(adcItem)
        bottomBar.addItem(supItem)


        // Click listener for the bottom bar buttons.
        bottomBar.setOnTabSelectedListener { position, wasSelected ->
            handleTabPress(position)
            true
        }
    }

    fun handleChangeInRole(itemId: Int): Boolean {
        val role = Role()
        return when(itemId) {
            R.id.action_top -> {
                view.setCorrectTabFragment(role.TOP)
                true
            }

            R.id.action_mid -> {
                view.setCorrectTabFragment(role.MID)
                true
            }

            R.id.action_jungle -> {
                view.setCorrectTabFragment(role.JUNGLE)
                true
            }

            R.id.action_support -> {
                view.setCorrectTabFragment(role.SUP)
                true
            }

            R.id.action_adc -> {
                view.setCorrectTabFragment(role.ADC)
                true
            }

            else -> {
                throw IllegalStateException("Incorrect role")
            }

        }
    }

    fun handleChangeInChamp() {
        // fetch champ from view

    }

    companion object {
        private val TOP = 0
        private val JUNGLE = 1
        private val MID = 2
        private val MARKSMAN = 3
        private val SUPPORT = 4
    }
}
